File: java-api/src/main/java/net/kfoundation/java/serialization/K4ObjectDeserializer.java
// --------------------------------------------------------------------------
//   ██╗  ██╗███████╗
//   ██║ ██╔╝██╔════╝   The KFoundation Project (www.kfoundation.net)
//   █████╔╝ █████╗     KFoundation for Scala Library
//   ██╔═██╗ ██╔══╝     Copyright (c) 2020 Mindscape Inc.
//   ██║  ██╗██║        Terms of KnoRBA Free Public License Agreement Apply
//   ╚═╝  ╚═╝╚═╝
// --------------------------------------------------------------------------

package net.kfoundation.java.serialization;

import java.io.InputStream;


/**
 * K4 object deserializer
 */
public class K4ObjectDeserializer extends WrappedScalaObjectDeserializer {
    public K4ObjectDeserializer(InputStream input) {
        super(net.kfoundation.scala.serialization.K4ObjectDeserializer
            .FACTORY().of(input));
    }
}

File: .js/src/main/scala/net/kfoundation/js/serialization/K4ObjectDeserializer.scala
package net.kfoundation.js.serialization

import net.kfoundation.scala.{serialization => s}

import scala.scalajs.js.annotation.JSExportTopLevel

@JSExportTopLevel("K4ObjectDeserializer")
class K4ObjectDeserializer(input: String) extends ObjectDeserializer {

  override def getImplementation: s.ObjectDeserializer =
    new s.K4ObjectDeserializer(input)

}
